Sources/Overlays/_Testing_Foundation/Support/Additions/BundleAdditions.swift

```diff
@@ -0,0 +1,61 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+#if canImport(Foundation)
+@_spi(ForSwiftTestingOnly) private import Testing
+public import Foundation
+
+extension Bundle {
+#if SWT_TARGET_OS_APPLE && !SWT_NO_DYNAMIC_LINKING && !SWT_NO_FILE_IO
+  /// Storage for ``testTarget``.
+  ///
+  /// On Apple platforms, the bundle containing test content is a loadable
+  /// XCTest bundle. By the time this property is read, the bundle should have
+  /// already been loaded.
+  private static let _testTarget: Bundle? = {
+    Test.testBundlePath.flatMap { imagePath in
+      // Construct a lazy sequence of URLs corresponding to the directories that
+      // contain the loaded image.
+      let imageURL = URL(fileURLWithFileSystemRepresentation: imagePath, isDirectory: false, relativeTo: nil)
+      let containingDirectoryURLs = sequence(first: imageURL) { url in
+        try? url.resourceValues(forKeys: [.parentDirectoryURLKey]).parentDirectory
+      }
+
+      // Find the directory most likely to contain our test content and return it.
+      return containingDirectoryURLs.lazy
+        .filter { $0.pathExtension.caseInsensitiveCompare("xctest") == .orderedSame }
+        .compactMap(Bundle.init(url:))
+        .first { _ in true }
+    }
+  }()
+#endif
+
+  /// A bundle representing the currently-running test target.
+  ///
+  /// On Apple platforms, this bundle represents the test bundle built by Xcode
+  /// or Swift Package Manager. On other platforms, it is equal to the main
+  /// bundle and represents the test executable built by Swift Package Manager.
+  ///
+  /// If more than one test bundle has been loaded into the current process, the
+  /// value of this property represents the first test bundle found by the
+  /// testing library at runtime.
+  ///
+  /// - Note: This property accesses the file system the first time it is used.
+  @_spi(Experimental)
+  public static var testTarget: Bundle {
```

Comment:
I do not think we should extend Foundation this way from Testing. The correct way would be implementation in Foundation itself, following the process there.

Reply:
This is a prototype only. 🙂 We can certainly follow the Foundation review process for this feature (in whatever form it takes) although due to layering it may need to live here in the overlay.

I have no plans to merge this without your team's signoff!